src/SqlPersistence/Installer.cs

```diff
@@ -4,53 +4,47 @@
     ReadOnlySettings settings;
     static ILog log = LogManager.GetLogger<Installer>();
 
     public Installer(ReadOnlySettings settings)
     {
         this.settings = settings;
+        installerSettings = settings.GetOrDefault<InstallerSettings>();
     }
 
     public async Task Install(string identity)
     {
-        //If neither of the features is configured then do not even attempt to validate config
-        if (!settings.ShouldInstall<SqlOutboxFeature>()
-            && !settings.ShouldInstall<SqlSubscriptionFeature>()
-            && !settings.ShouldInstall<SqlTimeoutFeature>()
-            && !settings.ShouldInstall<SqlSagaFeature>())
+        if (installerSettings == null || installerSettings.Disabled)
         {
             return;
         }
 
-        var connectionBuilder = settings.GetConnectionBuilder();
-        var sqlDialect = settings.GetSqlDialect();
-        var scriptDirectory = ScriptLocation.FindScriptDirectory(sqlDialect);
-        var tablePrefix = settings.GetTablePrefix();
+        installerSettings.Dialect.ValidateTablePrefix(installerSettings.TablePrefix);
 
-        sqlDialect.ValidateTablePrefix(tablePrefix);
-
-        using (var connection = await connectionBuilder.OpenConnection().ConfigureAwait(false))
+        using (var connection = await installerSettings.ConnectionBuilder.OpenConnection().ConfigureAwait(false))
         using (var transaction = connection.BeginTransaction())
         {
-            await InstallOutbox(scriptDirectory, connection, transaction, tablePrefix, sqlDialect).ConfigureAwait(false);
-            await InstallSagas(scriptDirectory, connection, transaction, tablePrefix, sqlDialect).ConfigureAwait(false);
-            await InstallSubscriptions(scriptDirectory, connection, transaction, tablePrefix, sqlDialect).ConfigureAwait(false);
-            await InstallTimeouts(scriptDirectory, connection, transaction, tablePrefix, sqlDialect).ConfigureAwait(false);
+            await InstallOutbox(installerSettings.ScriptDirectory, connection, transaction, installerSettings.TablePrefix, installerSettings.Dialect).ConfigureAwait(false);
+            await InstallSagas(installerSettings.ScriptDirectory, connection, transaction, installerSettings.TablePrefix, installerSettings.Dialect).ConfigureAwait(false);
+            await InstallSubscriptions(installerSettings.ScriptDirectory, connection, transaction, installerSettings.TablePrefix, installerSettings.Dialect).ConfigureAwait(false);
+            await InstallTimeouts(installerSettings.ScriptDirectory, connection, transaction, installerSettings.TablePrefix, installerSettings.Dialect).ConfigureAwait(false);
 
             transaction.Commit();
         }
     }
-
+    
     Task InstallOutbox(string scriptDirectory, DbConnection connection, DbTransaction transaction, string tablePrefix, SqlDialect sqlDialect)
     {
-        if (!settings.ShouldInstall<SqlOutboxFeature>())
+        if (!settings.IsFeatureActive(typeof(SqlOutboxFeature)))
```

Comment:
Just looking at this piece of code, without analyzing context in depth... It's a bit confusing, as you may have a feature active but don't want to execute installers (e.g. in prod). Maybe it's just a matter of naming, but I'll definitely stop here for a moment in the future to think.

Reply:
The "installers disabled for prod" condition is handled in the calling method. The check here is only to install only the bits that are required given the feature set configured in the endpoint e.g. if there is no sagas, do not run saga installers.